python/ray/_private/runtime_env/packaging.py

```diff
@@ -170,11 +173,9 @@ def parse_uri(pkg_uri: str) -> Tuple[Protocol, str]:
     if protocol == Protocol.S3 or protocol == Protocol.GS:
         return (protocol,
                 f"{protocol.value}_{uri.netloc}{uri.path.replace('/', '_')}")
-    elif protocol == Protocol.HTTPS:
-        return (
-            protocol,
-            f"https_{uri.netloc.replace('.', '_')}{uri.path.replace('/', '_')}"
-        )
+    elif protocol == Protocol.HTTPS or protocol == Protocol.HTTP:
+        cleaned_path_name = re.sub("[:/]", "_", uri.netloc)
```

Comment:
It looks like the `cleaned_path_name` is a cleaned version of just the `uri.netloc`. We should keep both the cleaned `uri.path` and the cleaned `uri.netloc` in the file name. Otherwise, two packages from the same remote Git provider will have a name collision.

In the cleaned `uri.path`, we need to make sure to substitute the `.` with a `_`.